```
2018-12-20T16:33:01.6267127Z  sympy/functions/combinatorial/tests/test_comb_factorials.py:test_factorial_Mod 
2018-12-20T16:33:01.6267174Z Traceback (most recent call last):
2018-12-20T16:33:01.6267547Z   File "/opt/hostedtoolcache/Python/2.7.15/x64/lib/python2.7/site-packages/sympy-1.4.dev0-py2.7.egg/sympy/functions/combinatorial/tests/test_comb_factorials.py", line 217, in test_factorial_Mod

assert Mod(factorial(153, evaluate=False), r) == Mod(factorial(153), r)
AssertionError

2018-12-20T16:33:01.6267767Z ________________________________________________________________________________
2018-12-20T16:33:01.6267819Z  sympy/functions/combinatorial/tests/test_comb_factorials.py:test_binomial_Mod _
2018-12-20T16:33:01.6267864Z Traceback (most recent call last):
2018-12-20T16:33:01.6268248Z   File "/opt/hostedtoolcache/Python/2.7.15/x64/lib/python2.7/site-packages/sympy-1.4.dev0-py2.7.egg/sympy/functions/combinatorial/tests/test_comb_factorials.py", line 447, in test_binomial_Mod
2018-12-20T16:33:01.6268317Z     assert Mod(binomial(156675, 4433, evaluate=False), p) == Mod(binomial(156675, 4433), p)
2018-12-20T16:33:01.6268365Z AssertionError
```

Not sure why 2.7 is giving this error.

--- a/sympy/core/mod.py
+++ b/sympy/core/mod.py
@@ -1,6 +1,6 @@
 from __future__ import print_function, division
 
-from sympy.core.numbers import nan
+from sympy.core.numbers import nan, Integer
 from .function import Function
 
 
@@ -62,7 +62,7 @@ class Mod(Function):
             except TypeError:
                 pass
             else:
-                if type(d) is int:
+                if type(d) is int or type(d) is long:
                     rv = p - d*q
                     if (rv*q < 0) == True:
                         rv += q

